acceptance/tests/direct_puppet/static_catalog_env_control.rb

```diff
@@ -0,0 +1,348 @@
+test_name "Environment control of static catalogs" 
+require 'json'
+
+@testroot = master.tmpdir(File.basename(__FILE__, '/*'))
+### @testroot = "/etc/puppetlabs"
+
+@coderoot = "#{@testroot}/code"
+
+@master_opts = {
+  'main' => {
+    'environmentpath' => "#{@coderoot}/environments",
+  },
+}
+
+@agent_result_dirs = {}
+@catalog_files = {}
+
+agents.each do |agent|
+  @agent_result_dirs[agent.hostname] = agent.tmpdir('results')
+  @catalog_files[agent.hostname] = "#{on(agent, puppet('config', 'print', 'client_datadir')).stdout.chomp}/catalog/#{agent.hostname}.json"
+end
+
+@manifest = <<MANIFEST
+File {
+  ensure => directory,
+  mode => "0755",
+}
+
+file { 
+  '#{@testroot}':;
+  '#{@coderoot}':;
+  '#{@coderoot}/environments':;
+  '#{@coderoot}/environments/production':;
+  '#{@coderoot}/environments/production/manifests':;
+  '#{@coderoot}/environments/production/modules':;
+  '#{@coderoot}/environments/production/modules/hello':;
+  '#{@coderoot}/environments/production/modules/hello/manifests':;
+  '#{@coderoot}/environments/production/modules/hello/files':;
+
+  '#{@coderoot}/environments/canary':;
+  '#{@coderoot}/environments/canary/manifests':;
+  '#{@coderoot}/environments/canary/modules':;
+  '#{@coderoot}/environments/canary/modules/can_hello':;
+  '#{@coderoot}/environments/canary/modules/can_hello/manifests':;
+  '#{@coderoot}/environments/canary/modules/can_hello/files':;
+  
+}
+
+file { '/etc/puppetlabs/puppetserver/conf.d/puppetserver.conf' :
```

Comment:
Is this needed? I assumed the puppetserver package would install a default puppetserver.conf, and then we use `modify_tk_config` [below](https://github.com/puppetlabs/puppet/pull/4733/files#diff-f59339e31c660460ee3ee91418e4fdd2R189) to add the `code-*-command`s.